src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt

```diff
@@ -0,0 +1,67 @@
+package org.rust.lang.core.completion
+
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.impl.RustPathElementImpl
+import org.rust.lang.core.psi.impl.RustUseGlobElementImpl
+import org.rust.lang.core.psi.util.parentOfType
+
+fun RustNamedElement.createLookupElement(): LookupElement {
+    return when (this) {
+        is RustFnItemElement -> LookupElementBuilder.createWithIcon(this)
+            .withLookupString(name ?: "")
+            .withTailText(parameters?.text ?: "()")
+            .withTypeText(retType?.type?.text ?: "()")
+        is RustImplMethodMemberElement -> LookupElementBuilder.createWithIcon(this)
+            .withLookupString(name ?: "")
+            .withTailText(parameters?.text ?: "()")
+            .withTypeText(retType?.type?.text ?: "()")
+        is RustTraitMethodMemberElement -> LookupElementBuilder.createWithIcon(this)
+            .withLookupString(name ?: "")
+            .withTailText(parameters?.text ?: "()")
+            .withTypeText(retType?.type?.text ?: "()")
+        is RustConstItemElement -> LookupElementBuilder.createWithIcon(this)
+            .withLookupString(name ?: "")
+            .withTypeText(type.text)
+        is RustStaticItemElement -> LookupElementBuilder.createWithIcon(this)
+            .withLookupString(name ?: "")
+            .withTypeText(type.text)
+        is RustPathElementImpl -> {
+            // re-export needs to be resolved
+            val resolved = reference.resolve()
+            resolved?.createLookupElement() ?: LookupElementBuilder.createWithIcon(this)
+        }
+        is RustUseGlobElement -> {
+            // There is a problem here with ambiguous getReference(): RustReference.
+            // Completely possible this is a compiler issue. Not sure why though.
+            val that = this as RustNamedElement
```

Comment:
I think this is a pretty reasonable error. `RustUseGlob` is no longer a `RustNamedElement`, so this branch is a dead code. See https://github.com/intellij-rust/intellij-rust/commit/fe072310d335e7a26c80e090f51c3d7d41fe0c66 and https://github.com/intellij-rust/intellij-rust/commit/6be06285388ce2f884e243d3cb426298f048d9fe for motivation for the change with NamedElements. 

It makes sense to check if some other branches are dead code as well. The one with `RutsPathElementImpl` seems like it.